components/org.wso2.carbon.identity.webfinger/src/test/java/org/wso2/carbon/identity/webfinger/builders/DefaultWebFingerRequestBuilderTest.java

```diff
@@ -0,0 +1,238 @@
+        vector.add(WebFingerConstants.REL);
+        vector.add(WebFingerConstants.RESOURCE);
+        Enumeration<String> params = vector.elements();
+        when(request.getParameterNames()).thenReturn(params);
+
+        try {
+            defaultWebFingerRequestBuilder.buildRequest(request);
+        } catch (WebFingerEndpointException e) {
+            assertEquals(e.getErrorCode(), WebFingerConstants.ERROR_CODE_INVALID_RESOURCE);
+        }
+    }
+
+    @Test
+    public  void testResourceEmpty() throws WebFingerEndpointException {
+        Vector<String> vector = new Vector<>();
+        vector.add(WebFingerConstants.REL);
+        vector.add(WebFingerConstants.RESOURCE);
+        Enumeration<String> params = vector.elements();
+        when(request.getParameterNames()).thenReturn(params);
+        when(request.getParameter(WebFingerConstants.RESOURCE)).thenReturn("9443/.well-known/webfinger");
+        when(request.getParameter(WebFingerConstants.REL)).thenReturn(rel);
+
+        try {
+            defaultWebFingerRequestBuilder.buildRequest(request);
+        } catch (WebFingerEndpointException e) {
+            assertEquals(e.getErrorMessage(), "Scheme of the resource cannot be empty");
+        }
+    }
+
+    @Test
+    public  void testUserInfoInvalid() throws WebFingerEndpointException {
+        Vector<String> vector = new Vector<>();
+        vector.add(WebFingerConstants.REL);
+        vector.add(WebFingerConstants.RESOURCE);
+        Enumeration<String> params = vector.elements();
+        when(request.getParameterNames()).thenReturn(params);
+        when(request.getParameter(WebFingerConstants.RESOURCE)).thenReturn(invalidAcctResource);
+        when(request.getParameter(WebFingerConstants.REL)).thenReturn(rel);
+
+        try {
+            defaultWebFingerRequestBuilder.buildRequest(request);
+        } catch (WebFingerEndpointException e) {
+            assertEquals(e.getErrorMessage(), "Invalid host value.");
+        }
+    }
+
+    @Test
+    public  void testUserInfo() throws WebFingerEndpointException, UserStoreException {
+        Vector<String> vector = new Vector<>();
+        vector.add(WebFingerConstants.REL);
+        vector.add(WebFingerConstants.RESOURCE);
+        Enumeration<String> params = vector.elements();
+        when(request.getParameterNames()).thenReturn(params);
+        when(request.getParameter(WebFingerConstants.RESOURCE)).thenReturn(acctResource);
+        when(request.getParameter(WebFingerConstants.REL)).thenReturn(rel);
+        mockStatic(WebFingerServiceComponentHolder.class);
+        when(WebFingerServiceComponentHolder.getRealmService()).thenReturn(realmService);
+        when(realmService.getTenantManager()).thenReturn(tenantManager);
+        when(tenantManager.getTenantId(any(String.class))).thenReturn(MultitenantConstants.SUPER_TENANT_ID);
+
+        WebFingerRequest webFingerRequest = defaultWebFingerRequestBuilder.buildRequest(request);
+        assertNotNull(webFingerRequest);
```

Comment:
Use appropriate messages whenever possible

Reply:
Also other than checking for null, can we assert on the content as well ? . It seems like this method never returns null.